src/test/java/org/pesmypetcare/webservice/controller/petmanager/PetMedicationRestControllerTest.java

```diff
@@ -0,0 +1,135 @@
+package org.pesmypetcare.webservice.controller.petmanager;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.pesmypetcare.webservice.entity.MedicationEntity;
+import org.pesmypetcare.webservice.service.MedicationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.BDDMockito.willDoNothing;
+import static org.mockito.BDDMockito.willReturn;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ExtendWith(MockitoExtension.class)
+public class PetMedicationRestControllerTest {
+    private static String jsonMedicationEntity;
+    private static String jsonField;
+    private static MedicationEntity MedicationEntity;
+    private static List<Map<String, Object>> MedicationEntityList;
+    private static String owner;
+    private static String petName;
+    private static String dateName;
+    private static String date;
+    private static String date2;
+    private static String field;
+    private static int value;
+    private static String urlBase;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private MedicationService service;
+
+    @BeforeAll
+    public static void setUp() {
+        jsonMedicationEntity = "{\n"
+                + "  \"duration\": 2,\n"
+                + "  \"periodicity\": 1\n"
+                + "}";
+        jsonField = "{\n"
+                + "  \"quantity\": 3.0\n"
+                + "} ";
+        MedicationEntity = new MedicationEntity();
+        MedicationEntityList = new ArrayList<>();
+        owner = "Manolo";
+        petName = "Canpeque2";
+        dateName = "2019-02-13T10:30:00%Cloroform";
+        date = "2019-02-13T10:30:00";
+        date2 = "2021-02-13T10:30:00";
+        field = "quantity";
+        value = 3;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['3' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)